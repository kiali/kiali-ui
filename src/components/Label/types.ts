export interface Labels {
  name: string;
  value: string;
}
