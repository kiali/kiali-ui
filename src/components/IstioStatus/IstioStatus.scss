#istio-status {
  &.pf-c-data-list {
    color: var(--pf-c-tooltip__content--Color);
    --pf-c-data-list--BorderTopWidth: 0;
    margin: 0;
    padding: 0;
  }

  .pf-c-data-list__item {
    --pf-c-data-list--BackgroundColor: var(--pf-c-tooltip__content--BackgroundColor);
    --pf-c-data-list__item-item--BorderTopWidth: 0;
  }

  .pf-c-data-list__item-row {
    --pf-c-data-list__item-row--PaddingRight: 0;
    --pf-c-data-list__item-row--PaddingLeft: 0;
  }

  .pf-c-data-list__cell {
    --pf-c-data-list__cell--PaddingTop: 0;
    --pf-c-data-list__cell-cell--PaddingTop: 0;
  }

  .pf-c-data-list__item-content {
    --pf-c-data-list__item-content--PaddingBottom: 0;
  }

  &.pf-c-list {
    --pf-c-content--ul--PaddingLeft: 0;
    --pf-c-content--ul--MarginTop: 0;
    --pf-c-content--ul--MarginLeft: 0;

    .pf-l-split {
      color: var(--pf-c-tooltip__content--Color);

      .pf-l-split__item {
        text-align: left;
      }
    }
  }
}
