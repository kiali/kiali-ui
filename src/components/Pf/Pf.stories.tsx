import '../../app/App.css';

import * as React from 'react';
import { storiesOf } from '@storybook/react';

import PFHeader from './PfHeader';
import PFContainerNavVertical from './PfContainerNavVertical';

const stories = storiesOf('Pf', module);

stories.add('PFContainerNavVertical', () => (
  <PFContainerNavVertical>
    <div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
      Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
    </div>
  </PFContainerNavVertical>
));

stories.add('PFHeader', () => (
  <PFContainerNavVertical>
    <PFHeader>
      <h2> Awesome header </h2>
    </PFHeader>
    <div>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur.
      Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum
    </div>
  </PFContainerNavVertical>
));
