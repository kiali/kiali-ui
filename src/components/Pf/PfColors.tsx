export enum PfColors {
  Black100 = '#fafafa',
  Black150 = '#f5f5f5',
  Black200 = '#ededed',
  Black300 = '#d1d1d1',
  Black400 = '#bbb',
  Black500 = '#8b8d8f',
  Black600 = '#72767b',
  Black700 = '#4d5258',
  Black800 = '#393f44',
  Black900 = '#292e34',
  Black1000 = '#030303',
  Blue50 = '#def3ff',
  Blue100 = '#bee1f4',
  Blue200 = '#7dc3e8',
  Blue300 = '#39a5dc',
  Blue400 = '#0088ce',
  Blue500 = '#00659c',
  Blue600 = '#004368',
  Blue700 = '#002235',
  Cyan100 = '#bedee1',
  Cyan200 = '#7dbdc3',
  Cyan300 = '#3a9ca6',
  Cyan400 = '#007a87',
  Cyan500 = '#005c66',
  Cyan600 = '#003d44',
  Cyan700 = '#001f22',
  Gold100 = '#fbeabc',
  Gold200 = '#f9d67a',
  Gold300 = '#f5c12e',
  Gold400 = '#f0ab00',
  Gold500 = '#b58100',
  Gold600 = '#795600',
  Gold700 = '#3d2c00',
  Green100 = '#cfe7cd',
  Green200 = '#9ecf99',
  Green300 = '#6ec664',
  Green400 = '#3f9c35',
  Green500 = '#2d7623',
  Green600 = '#1e4f18',
  Green700 = '#0f280d',
  LightBlue100 = '#beedf9',
  LightBlue200 = '#7cdbf3',
  LightBlue300 = '#35caed',
  LightBlue400 = '#00b9e4',
  LightBlue500 = '#008bad',
  LightBlue600 = '#005c73',
  LightBlue700 = '#002d39',
  LightGreen100 = '#e4f5bc',
  LightGreen200 = '#c8eb79',
  LightGreen300 = '#ace12e',
  LightGreen400 = '#92d400',
  LightGreen500 = '#6ca100',
  LightGreen600 = '#486b00',
  LightGreen700 = '#253600',
  Orange100 = '#fbdebf',
  Orange200 = '#f7bd7f',
  Orange300 = '#f39d3c',
  Orange400 = '#ec7a08',
  Orange500 = '#b35c00',
  Orange600 = '#773d00',
  Orange700 = '#3b1f00',
  Purple100 = '#c7bfff',
  Purple200 = '#a18fff',
  Purple300 = '#8461f7',
  Purple400 = '#703fec',
  Purple500 = '#582fc0',
  Purple600 = '#40199a',
  Purple700 = '#1f0066',
  Red100 = '#cc0000',
  Red200 = '#a30000',
  Red300 = '#8b0000',
  Red400 = '#470000',
  Red500 = '#2c0000',

  White = '#fff',
  Black = '#030303',

  Blue = '#0088ce', // Blue400
  Cyan = '#007a87', // Cyan400
  Gold = '#f0ab00', // Gold400
  Green = '#3f9c35', // Green400
  LightBlue = '#00b9e4', // LightBlue400
  LightGreen = '#92d400', // LightGreen400
  Orange = '#ec7a08', // Orange400
  Red = '#cc0000', // Red100

  // Kiali colors that use PF colors
  Gray = '#72767b' // Black600
}

export enum PFColorVars {
  DangerColor = 'var(--pf-global--danger-color--100)',
  WarningColor = 'var(--pf-global--warning-color--100)',
  SuccessColor = 'var(--pf-global--success-color--100)'
}
