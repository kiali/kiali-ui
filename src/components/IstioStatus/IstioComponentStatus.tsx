import * as React from 'react';
import { ComponentStatus, Status } from '../../types/IstioStatus';
import { IconType } from '@patternfly/react-icons/dist/js/createIcon';
import { PFAlertColor } from '../Pf/PfColors';
import { CheckCircleIcon, ExclamationCircleIcon, ExclamationTriangleIcon } from '@patternfly/react-icons';
import { Split, SplitItem } from '@patternfly/react-core';

type Props = {
  componentStatus: ComponentStatus;
};

export type ComponentIcon = {
  color: string;
  icon: IconType;
};

const ErrorCoreComponent: ComponentIcon = {
  color: PFAlertColor.Danger,
  icon: ExclamationCircleIcon
};

const ErrorAddonComponent: ComponentIcon = {
  color: PFAlertColor.Warning,
  icon: ExclamationTriangleIcon
};

const SuccessComponent: ComponentIcon = {
  color: PFAlertColor.Success,
  icon: CheckCircleIcon
};

// Mapping Valid-Core to Icon representation.
const validToIcon: { [valid: string]: ComponentIcon } = {
  'false-false': ErrorAddonComponent,
  'false-true': ErrorCoreComponent,
  'true-false': SuccessComponent,
  'true-true': SuccessComponent
};

const statusMsg = {
  [Status.NotFound]: 'Not found',
  [Status.Unhealthy]: 'Not healthy',
  [Status.Unreachable]: 'Unreachable'
};

class IstioComponentStatus extends React.Component<Props> {
  renderIcon = (status: Status, isCore: boolean) => {
    const compIcon = validToIcon[`${status === Status.Healthy}-${isCore}`];
    const IconComponent = compIcon.icon;
    return <IconComponent style={{ color: compIcon.color, marginTop: 5 }} />;
  };

  renderCells = () => {
    const comp = this.props.componentStatus;

    return [
      <Split key={'cell-status-icon-' + comp.name} gutter={'md'}>
        <SplitItem>{this.renderIcon(this.props.componentStatus.status, this.props.componentStatus.is_core)}</SplitItem>
        <SplitItem isFilled={true}>{comp.name}</SplitItem>
        <SplitItem>{statusMsg[comp.status]}</SplitItem>
      </Split>
    ];
  };

  render() {
    return this.renderCells();
  }
}

export default IstioComponentStatus;
