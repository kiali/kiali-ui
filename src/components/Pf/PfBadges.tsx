import { Badge, Tooltip, TooltipPosition } from '@patternfly/react-core';
import React, { CSSProperties } from 'react';
import { style } from 'typestyle';
import { PFColors } from './PfColors';

export type PFBadgeType = {
  badge: string;
  tt?: React.ReactFragment;
};

// PF Badges used by Kiali, keep alphabetized
// avoid duplicate badge letters, especially if they may appear on the same page
export const PFBadges = Object.freeze({
  App: { badge: 'A', tt: 'Application' } as PFBadgeType,
  Adapter: { badge: 'A', tt: 'Adapter' } as PFBadgeType,
  AttributeManifest: { badge: 'AM', tt: 'Attribute Manifest' } as PFBadgeType,
  AuthorizationPolicy: { badge: 'AP', tt: 'Authorization Policy' } as PFBadgeType,
  Cluster: { badge: 'CL', tt: 'Cluster' } as PFBadgeType,
  ClusterRBACConfig: { badge: 'CRC', tt: 'Cluster RBAC Configuration' } as PFBadgeType,
  Container: { badge: 'C', tt: 'Container' } as PFBadgeType,
  DestinationRule: { badge: 'DR', tt: 'Destination Rule' } as PFBadgeType,
  EnvoyFilter: { badge: 'EF', tt: 'Envoy Filter' } as PFBadgeType,
  ExternalService: { badge: 'ES', tt: 'External Service' } as PFBadgeType,
  FederatedService: { badge: 'FS', tt: 'Federated Service' } as PFBadgeType,
  Gateway: { badge: 'G', tt: 'Gateway' } as PFBadgeType,
  Handler: { badge: 'H', tt: 'Handler' },
  Host: { badge: 'H', tt: 'Host' },
  Instance: { badge: 'I', tt: 'Instance' },
  MeshPolicy: { badge: 'MP', tt: 'Mesh Policy' } as PFBadgeType,
  Namespace: { badge: 'NS', tt: 'Namespace' } as PFBadgeType,
  Operation: { badge: 'O', tt: 'Operation' } as PFBadgeType,
  PeerAuthentication: { badge: 'PA', tt: 'Peer Authentication' } as PFBadgeType,
  Pod: { badge: 'P', tt: 'Pod' } as PFBadgeType,
  Policy: { badge: 'P', tt: 'Policy' } as PFBadgeType,
  RBACConfig: { badge: 'RC', tt: 'RBAC Configuration' } as PFBadgeType,
  RequestAuthentication: { badge: 'RA', tt: 'Request Authentication' } as PFBadgeType,
  Rule: { badge: 'R', tt: 'Rule' } as PFBadgeType,
  Service: { badge: 'S', tt: 'Service' } as PFBadgeType,
  ServiceEntry: { badge: 'SE', tt: 'Service Entry' } as PFBadgeType,
  ServiceRole: { badge: 'SR', tt: 'Service Role' } as PFBadgeType,
  ServiceRoleBinding: { badge: 'SRB', tt: 'Service Role Binding' } as PFBadgeType,
  Sidecar: { badge: 'SC', tt: 'Istio Sidecar Proxy' } as PFBadgeType,
  Template: { badge: 'T', tt: 'Template' } as PFBadgeType,
  Unknown: { badge: 'U', tt: 'Unknown' } as PFBadgeType,
  VirtualService: { badge: 'VS', tt: 'Virtual Service' } as PFBadgeType,
  Workload: { badge: 'W', tt: 'Workload' } as PFBadgeType,
  WorkloadEntry: { badge: 'WE', tt: 'Workload Entry' } as PFBadgeType,
  WorkloadGroup: { badge: 'WG', tt: 'Workload Group' } as PFBadgeType
});

export const kialiBadge = style({
  backgroundColor: PFColors.Badge,
  borderRadius: '30em',
  marginRight: '10px'
});

type PFBadgeProps = {
  badge: PFBadgeType;
  className?: string; // default=kialiBadge
  id?: string;
  isRead?: boolean;
  key?: string;
  position?: TooltipPosition; // default=auto
  style?: CSSProperties;
};

export class PFBadge extends React.PureComponent<PFBadgeProps> {
  render() {
    const key = `pfbadge-${this.props.badge.badge}`;
    const badge = (
      <Badge
        className={this.props.className || kialiBadge}
        id={this.props.id || key}
        isRead={this.props.isRead || false}
        key={this.props.key || key}
        style={this.props.style}
      >
        {this.props.badge.badge}
      </Badge>
    );

    return !this.props.badge.tt ? (
      badge
    ) : (
      <Tooltip
        content={<>{this.props.badge.tt}</>}
        id={`tt-${this.props.id || key}`}
        key={`tt-${this.props.key || key}`}
        position={this.props.position || TooltipPosition.auto}
      >
        {badge}
      </Tooltip>
    );
  }
}

export const getPFBadge = (badge: string, tt?: React.ReactFragment): React.ReactFragment => {
  return <PFBadge badge={{ badge: badge, tt: tt }} />;
};
