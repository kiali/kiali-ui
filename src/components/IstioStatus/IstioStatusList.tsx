import * as React from 'react';
import { List, Text, TextContent, TextVariants } from '@patternfly/react-core';
import { ComponentStatus, Status } from '../../types/IstioStatus';
import IstioComponentStatus from './IstioComponentStatus';
import { PfColors } from '../Pf/PfColors';

type Props = {
  status: ComponentStatus[];
};

class IstioStatusList extends React.Component<Props> {
  nonhealthyComponents = () => {
    return this.props.status.filter((c: ComponentStatus) => c.status !== Status.Healthy);
  };

  coreComponentsStatus = () => {
    return this.nonhealthyComponents().filter((s: ComponentStatus) => s.is_core);
  };

  addonComponentsStatus = () => {
    return this.nonhealthyComponents().filter((s: ComponentStatus) => !s.is_core);
  };

  renderComponentList = () => {
    const groups = {
      core: this.coreComponentsStatus,
      addon: this.addonComponentsStatus
    };

    return ['core', 'addon'].map((group: string) => {
      return (
        <>
          {groups[group]().map(status => {
            return <IstioComponentStatus componentStatus={status} />;
          })}
        </>
      );
    });
  };

  render() {
    return (
      <TextContent style={{ color: PfColors.White }}>
        <Text component={TextVariants.h4}>Istio Components Status</Text>
        <List id="istio-status" aria-label="Istio Component List">
          {this.renderComponentList()}
        </List>
      </TextContent>
    );
  }
}

export default IstioStatusList;
