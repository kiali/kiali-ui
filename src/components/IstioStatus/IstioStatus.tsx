import * as React from 'react';
import * as API from '../../services/Api';
import * as AlertUtils from '../../utils/AlertUtils';
import { TimeInMilliseconds } from '../../types/Common';
import { ComponentStatus, Status } from '../../types/IstioStatus';
import { MessageType } from '../../types/MessageCenter';
import Namespace from '../../types/Namespace';
import { KialiAppState } from '../../store/Store';
import { istioStatusSelector, lastRefreshAtSelector, namespaceItemsSelector } from '../../store/Selectors';
import { KialiDispatch } from '../../types/Redux';
import { bindActionCreators } from 'redux';
import { IstioStatusActions } from '../../actions/IstioStatusActions';
import { connect } from 'react-redux';
import { Tooltip, TooltipPosition } from '@patternfly/react-core';
import IstioStatusList from './IstioStatusList';
import { PFAlertColor } from '../Pf/PfColors';
import './IstioStatus.css';
import { ResourcesFullIcon } from '@patternfly/react-icons';

type ReduxProps = {
  lastRefreshAt: TimeInMilliseconds;
  setIstioStatus: (istioStatus: ComponentStatus[]) => void;
  namespaces: Namespace[] | undefined;
  status: ComponentStatus[];
};

type Props = ReduxProps & {};

const ValidToColor = {
  'false-false': PFAlertColor.Danger,
  'false-true': PFAlertColor.Danger,
  'true-false': PFAlertColor.Warning,
  'true-true': PFAlertColor.Success
};

export class IstioStatus extends React.Component<Props> {
  componentDidMount() {
    this.fetchStatus();
  }

  componentDidUpdate(prevProps: Readonly<Props>): void {
    if (this.props.lastRefreshAt !== prevProps.lastRefreshAt) {
      this.fetchStatus();
    }
  }

  fetchStatus = () => {
    API.getIstioStatus()
      .then(response => {
        return this.props.setIstioStatus(response.data);
      })
      .catch(error => {
        // User without namespaces can't have access to mTLS information. Reduce severity to info.
        const informative = this.props.namespaces && this.props.namespaces.length < 1;
        if (informative) {
          AlertUtils.addError('Istio deployment status disabled.', error, 'default', MessageType.INFO);
        } else {
          AlertUtils.addError('Error fetching Istio deployment status.', error, 'default', MessageType.ERROR);
        }
      });
  };

  tooltipContent = () => {
    return <IstioStatusList status={this.props.status} />;
  };

  tooltipColor = () => {
    let coreHealthy: boolean = true;
    let addonHealthy: boolean = true;

    Object.keys(this.props.status || {}).forEach((compKey: string) => {
      const { status, is_core } = this.props.status[compKey];
      const isHealthy: boolean = status === Status.Healthy;

      if (is_core) {
        coreHealthy = coreHealthy && isHealthy;
      } else {
        addonHealthy = addonHealthy && isHealthy;
      }
    });

    return ValidToColor[`${coreHealthy}-${addonHealthy}`];
  };

  healthyComponents = () => {
    return this.props.status.reduce((healthy: boolean, compStatus: ComponentStatus) => {
      return healthy && compStatus.status === Status.Healthy;
    }, true);
  };

  render() {
    if (!this.healthyComponents()) {
      return (
        <Tooltip position={TooltipPosition.left} enableFlip={true} content={this.tooltipContent()} maxWidth={'25rem'}>
          <ResourcesFullIcon color={this.tooltipColor()} style={{ verticalAlign: '-0.3em', marginRight: 8 }} />
        </Tooltip>
      );
    }

    return null;
  }
}

const mapStateToProps = (state: KialiAppState) => ({
  status: istioStatusSelector(state),
  lastRefreshAt: lastRefreshAtSelector(state),
  namespaces: namespaceItemsSelector(state)
});

const mapDispatchToProps = (dispatch: KialiDispatch) => ({
  setIstioStatus: bindActionCreators(IstioStatusActions.setinfo, dispatch)
});

const IstioStatusConnected = connect(
  mapStateToProps,
  mapDispatchToProps
)(IstioStatus);

export default IstioStatusConnected;
