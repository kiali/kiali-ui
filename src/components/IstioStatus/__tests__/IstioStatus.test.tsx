import * as React from 'react';
import { shallow } from 'enzyme';
import { ComponentStatus, Status } from '../../../types/IstioStatus';
import { IstioStatus } from '../IstioStatus';
import { shallowToJson } from 'enzyme-to-json';
import { PFAlertColor } from '../../Pf/PfColors';

const mockIcon = (componentList: ComponentStatus[]) => {
  return shallow(
    <IstioStatus
      status={componentList}
      lastRefreshAt={848152}
      namespaces={[{ name: 'bookinfo' }, { name: 'istio-system' }]}
      setIstioStatus={jest.fn()}
    />
  );
};

const testSnapshot = (wrapper: any) => {
  expect(shallowToJson(wrapper)).toBeDefined();
  expect(shallowToJson(wrapper)).toMatchSnapshot();
};

const testTooltip = (wrapper: any) => {
  // Testing the tooltip
  expect(wrapper.name()).toEqual('Tooltip');
  expect(wrapper.props().position).toEqual('left');
  expect(wrapper.props().enableFlip).toEqual(true);

  expect(wrapper.children().length).toEqual(1);
};

const testIcon = (wrapper: any, color: any) => {
  // Icon shown - should be red
  const icon = wrapper.childAt(0);
  expect(icon).toBeDefined();
  expect(icon.name()).toEqual('ResourcesFullIcon');
  expect(icon.props().color).toEqual(color);
};

describe('When core component has a problem', () => {
  it('the Icon shows is displayed in Red', () => {
    const wrapper = mockIcon([
      {
        name: 'grafana',
        status: Status.Healthy,
        is_core: false
      },
      {
        name: 'istio-egressgateway',
        status: Status.Unhealthy,
        is_core: true
      }
    ]);

    testSnapshot(wrapper);
    testTooltip(wrapper);
    testIcon(wrapper, PFAlertColor.Danger);
  });
});

describe('When addon component has a problem', () => {
  it('the Icon shows is displayed in orange', () => {
    const wrapper = mockIcon([
      {
        name: 'grafana',
        status: Status.Unhealthy,
        is_core: false
      },
      {
        name: 'istio-egressgateway',
        status: Status.Healthy,
        is_core: true
      }
    ]);

    testSnapshot(wrapper);
    testTooltip(wrapper);
    testIcon(wrapper, PFAlertColor.Warning);
  });
});

describe('When both core and addon component have problems', () => {
  it('the Icon shows is displayed in red', () => {
    const wrapper = mockIcon([
      {
        name: 'grafana',
        status: Status.Unhealthy,
        is_core: false
      },
      {
        name: 'istio-egressgateway',
        status: Status.Unhealthy,
        is_core: true
      }
    ]);

    testSnapshot(wrapper);
    testTooltip(wrapper);
    testIcon(wrapper, PFAlertColor.Danger);
  });
});

describe('When all components are good', () => {
  it('the Icon shows is displayed in green', () => {
    const wrapper = mockIcon([
      {
        name: 'grafana',
        status: Status.Healthy,
        is_core: false
      },
      {
        name: 'istio-egressgateway',
        status: Status.Healthy,
        is_core: true
      }
    ]);

    expect(wrapper.isEmptyRender()).toBeTruthy();
  });
});
