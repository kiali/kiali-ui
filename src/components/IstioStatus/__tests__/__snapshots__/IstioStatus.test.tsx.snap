// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When addon component has a problem the Icon shows is displayed in orange 1`] = `
<Tooltip
  appendTo={[Function]}
  aria="describedby"
  boundary="window"
  className=""
  content={
    <IstioStatusList
      status={
        Array [
          Object {
            "is_core": false,
            "name": "grafana",
            "status": "Unhealthy",
          },
          Object {
            "is_core": true,
            "name": "istio-egressgateway",
            "status": "Healthy",
          },
        ]
      }
    />
  }
  distance={15}
  enableFlip={true}
  entryDelay={500}
  exitDelay={500}
  flipBehavior={
    Array [
      "top",
      "right",
      "bottom",
      "left",
      "top",
      "right",
      "bottom",
    ]
  }
  id=""
  isAppLauncher={false}
  isContentLeftAligned={false}
  isVisible={false}
  maxWidth="25rem"
  position="left"
  tippyProps={Object {}}
  trigger="mouseenter focus"
  zIndex={9999}
>
  <ResourcesFullIcon
    color="var(--pf-global--warning-color--100)"
    noVerticalAlign={false}
    size="sm"
    style={
      Object {
        "marginRight": 8,
        "verticalAlign": "-0.3em",
      }
    }
    title={null}
  />
</Tooltip>
`;

exports[`When both core and addon component have problems the Icon shows is displayed in red 1`] = `
<Tooltip
  appendTo={[Function]}
  aria="describedby"
  boundary="window"
  className=""
  content={
    <IstioStatusList
      status={
        Array [
          Object {
            "is_core": false,
            "name": "grafana",
            "status": "Unhealthy",
          },
          Object {
            "is_core": true,
            "name": "istio-egressgateway",
            "status": "Unhealthy",
          },
        ]
      }
    />
  }
  distance={15}
  enableFlip={true}
  entryDelay={500}
  exitDelay={500}
  flipBehavior={
    Array [
      "top",
      "right",
      "bottom",
      "left",
      "top",
      "right",
      "bottom",
    ]
  }
  id=""
  isAppLauncher={false}
  isContentLeftAligned={false}
  isVisible={false}
  maxWidth="25rem"
  position="left"
  tippyProps={Object {}}
  trigger="mouseenter focus"
  zIndex={9999}
>
  <ResourcesFullIcon
    color="var(--pf-global--danger-color--100)"
    noVerticalAlign={false}
    size="sm"
    style={
      Object {
        "marginRight": 8,
        "verticalAlign": "-0.3em",
      }
    }
    title={null}
  />
</Tooltip>
`;

exports[`When core component has a problem the Icon shows is displayed in Red 1`] = `
<Tooltip
  appendTo={[Function]}
  aria="describedby"
  boundary="window"
  className=""
  content={
    <IstioStatusList
      status={
        Array [
          Object {
            "is_core": false,
            "name": "grafana",
            "status": "Healthy",
          },
          Object {
            "is_core": true,
            "name": "istio-egressgateway",
            "status": "Unhealthy",
          },
        ]
      }
    />
  }
  distance={15}
  enableFlip={true}
  entryDelay={500}
  exitDelay={500}
  flipBehavior={
    Array [
      "top",
      "right",
      "bottom",
      "left",
      "top",
      "right",
      "bottom",
    ]
  }
  id=""
  isAppLauncher={false}
  isContentLeftAligned={false}
  isVisible={false}
  maxWidth="25rem"
  position="left"
  tippyProps={Object {}}
  trigger="mouseenter focus"
  zIndex={9999}
>
  <ResourcesFullIcon
    color="var(--pf-global--danger-color--100)"
    noVerticalAlign={false}
    size="sm"
    style={
      Object {
        "marginRight": 8,
        "verticalAlign": "-0.3em",
      }
    }
    title={null}
  />
</Tooltip>
`;
